sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs

```diff
@@ -556,5 +556,201 @@ public static RecognizeLinkedEntitiesResultCollection RecognizeLinkedEntitiesRes
         }
 
         #endregion Linked Entities
+
+        #region Action Result Models
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.AnalyzeActionsResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="extractKeyPhrasesActionResult">Sets the collection of <see cref="TextAnalytics.ExtractKeyPhrasesActionResult"/> property.</param>
+        /// <param name="recognizeEntitiesActionResults">Sets the collection of <see cref="TextAnalytics.RecognizeEntitiesActionResult"/> property.</param>
+        /// <param name="recognizePiiEntitiesActionResults">Sets the collection of <see cref="RecognizePiiEntitiesActionResult"/> property.</param>
+        /// <param name="recognizeLinkedEntitiesActionsResults">Sets the collection of <see cref="TextAnalytics.RecognizeLinkedEntitiesActionResult"/> property.</param>
+        /// <param name="analyzeSentimentActionsResults">Sets the collection of <see cref="TextAnalytics.AnalyzeSentimentActionResult"/> property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.AnalyzeActionsResult"/> for mocking purposes.</returns>
+        public static AnalyzeActionsResult AnalyzeActionsResult(
+            IReadOnlyCollection<ExtractKeyPhrasesActionResult> extractKeyPhrasesActionResult,
+            IReadOnlyCollection<RecognizeEntitiesActionResult> recognizeEntitiesActionResults,
+            IReadOnlyCollection<RecognizePiiEntitiesActionResult> recognizePiiEntitiesActionResults,
+            IReadOnlyCollection<RecognizeLinkedEntitiesActionResult> recognizeLinkedEntitiesActionsResults,
+            IReadOnlyCollection<AnalyzeSentimentActionResult> analyzeSentimentActionsResults)
+        {
+            return new AnalyzeActionsResult(extractKeyPhrasesActionResult, recognizeEntitiesActionResults, recognizePiiEntitiesActionResults, recognizeLinkedEntitiesActionsResults, analyzeSentimentActionsResults);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="TextAnalytics.ExtractKeyPhrasesActionResult"/> for mocking purposes.
+        /// </summary>
+        /// <param name="result">Sets the <see cref="TextAnalytics.ExtractKeyPhrasesResultCollection"/> property.</param>
+        /// <param name="completedOn">Sets the <see cref="DateTimeOffset"/> property.</param>
+        /// <param name="code">Sets the code property.</param>
+        /// <param name="message">Sets the message property.</param>
+        /// <returns>A new instance of <see cref="TextAnalytics.ExtractKeyPhrasesActionResult"/> for mocking purposes.</returns>
+        public static ExtractKeyPhrasesActionResult ExtractKeyPhrasesActionResult(
```

Comment:
All the`xxActionResult` types should have two methods. One in case the action succeded and one in case the action failed. For example, this is similar to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsModelFactory.cs#L171-L192

Reply:
@maririos 
but the models they call only have one constructor that contains both parameters and error
in the example you're referring to, the model being called has 2 constructors, one in case of success, and one in case of failure!
unlike the models i'm dealing with
what i'm thinking is, create two constructors for these models, to have one for success, and one for failure
but this would impact the existing code that already uses the old constructor
let me know what you think

Reply:
I see. Creating the two constructors sounds good. Your can make them internal so they are still not exposed for the user and add comments about how these methods are only intended to use for mocking purposes